libs/core/type_support/include/hpx/type_support/is_contiguous_iterator.hpp

```diff
@@ -0,0 +1,63 @@
+//  Copyright (c) 2007-2022 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace hpx::traits {
+    ///////////////////////////////////////////////////////////////////////////
+    // Iterators are contiguous if they are pointers (without concepts we have
+    // no generic way of determining whether an iterator is contiguous)
+
+    namespace detail {
+
+        // Iterators returned from std::vector are contiguous (by definition)
+
+        // different versions of clang-format disagree
+        template <typename Iter,
+            typename T = typename std::iterator_traits<Iter>::value_type>
+        // clang-format off
+        struct is_vector_iterator
+          : std::integral_constant<bool,
+                std::is_same_v<
+                    decltype(std::declval<std::vector<T>&>().begin()), Iter> ||
+                std::is_same_v<
+                    decltype(std::declval<std::vector<T>&>().cbegin()), Iter>>
+        // clang-format on
+        {
+        };
+    }    // namespace detail
```

Comment:
Should we add the iterators exposed by `std::array` and `hpx::detail::small_vector` to the mix, possibly others?

Reply:
Yes, totally